cmd/posix.go

```diff
@@ -100,22 +99,21 @@ func isDirEmpty(dirname string) bool {
 }
 
 // Initialize a new storage disk.
-func newPosix(diskPath string) (StorageAPI, error) {
-	if diskPath == "" {
+func newPosix(path string) (StorageAPI, error) {
+	if path == "" {
 		return nil, errInvalidArgument
 	}
-	suppliedDiskPath := diskPath
-	var err error
+	diskPath := path
 	// Disallow relative paths, figure out absolute paths.
+	var err error
 	diskPath, err = filepath.Abs(diskPath)
```

Comment:
you can do `diskPath, err := filepath.Abs(diskPath)` and delete 2 lines above it